packages/fetcher/index.ts

```diff
@@ -34,12 +34,18 @@ export abstract class Fetcher<C> {
    */
   config: C;
 
+  /**
+   * The logger to use for logging
+   */
+  logger?: Logger;
+
   /**
    * Initialize the fetcher
    * @param config The configuration for the Fetcher
    */
-  constructor(config: C) {
+  constructor(config: C, logger?: Logger) {
```

Comment:
Is this to provide the user a way of disabling logging altogether, or are we making this optional for backwards compatibility perhaps?

Reply:
My focus was the former (make logging optional so you can disable it altogether) but the latter plays nicely into that 🙂